In [81]:
import pandas as pd
import matplotlib.pyplot as plt
import numpy as np
import statistics

data = pd.DataFrame()
data["name"] = ["Greg", "Marcia", "Peter", "Jan", "Bobby", "Cindy", "Oliver"]
data["age"] = [14, 12, 11, 10, 8, 6, 8]
print(data) 

     name  age
0    Greg   14
1  Marcia   12
2   Peter   11
3     Jan   10
4   Bobby    8
5   Cindy    6
6  Oliver    8


In [82]:
data.describe()

age
count   7.000000
mean    9.857143
std     2.734262
min     6.000000
25%     8.000000
50%    10.000000
75%    11.500000
max    14.000000

In [83]:
mean = np.mean(data["age"])
median = np.median(data["age"])
mode = statistics.mode(data["age"])
v = np.var(data.age)
s = np.std(data["age"], ddof=1)
se = s/np.sqrt(len(data["age"]-1))

print("mean: " + str(mean))
print("median: " + str(median))
print("mode: " + str(mode))
print("varience: " + str(v))
print("standard deviation: " + str(s))
print("standard error: " + str(se))

mean: 9.857142857142858
median: 10.0
mode: 8
varience: 6.408163265306122
standard deviation: 2.734262327610589
standard error: 1.0334540197243192


Using these estimates, if you had to choose only one estimate of central tendency and one estimate of variance to describe the data, which would you pick and why?

My answer: 

In [84]:
data["age"] = [14, 12, 11, 10, 8, 7, 8]

In [85]:
mean = np.mean(data["age"])
median = np.median(data["age"])
mode = statistics.mode(data["age"])
v = np.var(data.age)
s = np.std(data["age"], ddof=1)
se = s/np.sqrt(len(data["age"]-1))

print("mean: " + str(mean))
print("median: " + str(median))
print("mode: " + str(mode))
print("varience: " + str(v))
print("standard deviation: " + str(s))
print("standard error: " + str(se))

mean: 10.0
median: 10.0
mode: 8
varience: 5.428571428571429
standard deviation: 2.516611478423583
standard error: 0.9511897312113418


Changed: mean(9.85=>10.0), variance(6.40=>5.42) , standard deviation(2.73=>2.51), standard error(1.03=>0.95)

Not Changed: median, mode

Reason: Cindy is the youngest. So, when her age increased, it reduces the variance and caused changes in S and SE as well.

In [87]:
print(data)
data["name"][6] = "Jessica"
data["age"][6] = 1
print(data)

      name  age
0     Greg   14
1   Marcia   12
2    Peter   11
3      Jan   10
4    Bobby    8
5    Cindy    7
6  Jessica    9
      name  age
0     Greg   14
1   Marcia   12
2    Peter   11
3      Jan   10
4    Bobby    8
5    Cindy    7
6  Jessica    1


//anaconda3/lib/python3.7/site-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  
//anaconda3/lib/python3.7/site-packages/ipykernel_launcher.py:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  This is separate from the ipykernel package so we can avoid doing imports until


In [89]:
mean = np.mean(data["age"])
median = np.median(data["age"])
v = np.var(data.age)
s = np.std(data["age"], ddof=1)
se = s/np.sqrt(len(data["age"]-1))

print("mean: " + str(mean))
print("median: " + str(median))
print("mode: None")
print("varience: " + str(v))
print("standard deviation: " + str(s))
print("standard error: " + str(se))

mean: 9.0
median: 10.0
mode: None
varience: 15.428571428571429
standard deviation: 4.242640687119285
standard error: 1.6035674514745462
